server/internal/infrastructure/s3/aws_s3.go

```diff
@@ -0,0 +1,344 @@
+package s3
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"path"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/kennygrant/sanitize"
+	"github.com/reearth/reearth/server/internal/usecase/gateway"
+	"github.com/reearth/reearth/server/pkg/file"
+	"github.com/reearth/reearth/server/pkg/id"
+	"github.com/reearth/reearthx/log"
+	"github.com/reearth/reearthx/rerror"
+	"github.com/samber/lo"
+)
+
+const (
+	s3AssetBasePath  string = "assets"
+	s3PluginBasePath string = "plugins"
+	s3MapBasePath    string = "maps"
+	s3FileSizeLimit  int64  = 1024 * 1024 * 100 // about 100MB
```

Comment:
```suggestion
	assetBasePath  string = "assets"
	pluginBasePath string = "plugins"
	mapBasePath    string = "maps"
	fileSizeLimit  int64  = 1024 * 1024 * 100 // about 100MB
```
Since the package is independent, there is no conflict in variable names with the GCS implementation.